### ISSUE TITLE
[BUG] Exception if there's a root path in spec

### ISSUE THREAD

**Issue by VentyCZ** [mtennoe/swagger-typescript-codegen#108]
Just tried to integrate this library into my project and I just found a case where the library fails.

There is an issue with the [getPathToMethodName](https://github.com/mtennoe/swagger-typescript-codegen/blob/4cb5e72cbbb2b895472d2ddf1ffaefb9e49b0268/src/view-data/method.ts#L100) function.

When I have a root path (**/**) in the paths object of the spec, the library crashes: "Cannot read property 'toUpperCase' of undefined".

It's caused by line 116, but originates from line above.

If the result variable is an empty string it throws.

I "fixed" it by doing this before calling the getTypescriptCode function:
```js
if (spec.paths.hasOwnProperty('/')) {
    let prop = spec.paths['/'];
    delete spec.paths['/'];
    spec.paths['/index'] = prop;
}
```
This solution fixes the issue partially, because now the target path in the generated code is wrong.

The solution to the issue might be using an alias for the / path, like index for example.

**Comment by sjwilczynski** [comment]
Created a [PR](https://github.com/mtennoe/swagger-typescript-codegen/pull/109) for that.

**Comment by VentyCZ** [comment]
@sjwilczynski Thanks for the fix! So it will generate methods like get_root_(), right ?

**Comment by mtennoe** [comment]
Thanks for reporting @VentyCZ, and thanks for the PR @sjwilczynski  - Will take a look today

**Comment by mtennoe** [comment]
Merged and published as 3.0.3. Thanks @sjwilczynski!

**Comment by sjwilczynski** [comment]
> @sjwilczynski Thanks for the fix! So it will generate methods like get_root_(), right ?

After some changes we decided on methods names like `rootEndpoint_${lowerCasedHttpVerb}` - the http method comes after so that there is no conflict when someone has an endpoint called "rootEnpoint"
